ofrak_core/ofrak/service/serialization/serializers/data_service_serializer.py

```diff
@@ -19,71 +49,33 @@ class DataServiceSerializer(SerializerInterface):
 
     targets = (DataService,)
 
+    data_service_annotations = {
+        "_model_store": Dict[DataId, DataModel],
+        "_roots": Dict[DataId, _DataRoot],
+    }
+
     def obj_to_pjson(self, obj: DataService, _type_hint: Any) -> Dict[str, PJSONType]:
         data_service_pjson = {
-            "_model_store": self._service.to_pjson(obj._model_store, Dict[bytes, DataModel]),
-            "_roots": [self._data_root_to_pjson(data_root) for data_root in obj._roots.values()],
+            attr_name: self._service.to_pjson(getattr(obj, attr_name), type_hint)
+            for attr_name, type_hint in self.data_service_annotations.items()
         }
-
         return data_service_pjson
 
     def pjson_to_obj(self, pjson_obj: Dict[str, Dict], _type_hint: Any) -> DataService:
-        _model_store = self._service.from_pjson(pjson_obj["_model_store"], Dict[bytes, DataModel])
-        _roots = dict()
-        for root_pjson in pjson_obj["_roots"]:
-            root_obj = self._data_root_from_pjson(root_pjson, _model_store)
-            _roots[root_obj.model.id] = root_obj
-
+        deserialized_attrs = {
+            attr_name: self._service.from_pjson(pjson_obj[attr_name], type_hint)
+            for attr_name, type_hint in self.data_service_annotations.items()
+        }
+        model_store = deserialized_attrs["_model_store"]
+        roots = deserialized_attrs["_roots"]
+        # The DataService assumes that the DataModel instance in the model store is the same
```

Comment:
I was going to leave a comment in the _Waypoint deserializer pointing this out, then thought I should scroll ahead and see if you thought of it already. Nice :)

However, doesn't this make serializing/deserializing the `model_store` completely redundant now? Everything in there is either a root or child of a root and replaced here, so handling `model_store` is just wasted time and space.

Reply:
Good point! I've removed that from serialization.